infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -272,26 +272,33 @@ private LogReplicationEntry processReceivedMessage(LogReplicationEntry message)
             return logEntrySinkBufferManager.processMsgAndBuffer(message);
         } else {
             LogReplicationEntry ack = snapshotSinkBufferManager.processMsgAndBuffer(message);
-            // Check to the one persisted...
-            if (ack.getMetadata().getMessageMetadataType() == MessageType.SNAPSHOT_REPLICATED) {
-                long lastAppliedBaseSnapshotTimestamp = logReplicationMetadataManager.getLastAppliedBaseSnapshotTimestamp();
-                long latestSnapshotSyncCycleId = logReplicationMetadataManager.getCurrentSnapshotSyncCycleId();
-                long ackSnapshotSyncCycleId = ack.getMetadata().getSyncRequestId().getMostSignificantBits() & Long.MAX_VALUE;
-                // Verify this snapshot ACK corresponds to the last initialized/valid snapshot sync
-                // as a previous one could have been canceled but still processed due to messages being out of order
-                if ((ackSnapshotSyncCycleId == latestSnapshotSyncCycleId) &&
-                        (ack.getMetadata().getSnapshotTimestamp() == lastAppliedBaseSnapshotTimestamp)) {
-                    // Notify end of snapshot sync. This is a blocking call.
-                    snapshotSyncPlugin.onSnapshotSyncEnd(runtime);
-                } else {
-                    log.warn("SNAPSHOT_SYNC has completed for {}, but new ongoing SNAPSHOT_SYNC is {}",
-                            ack.getMetadata().getSnapshotTimestamp(), lastAppliedBaseSnapshotTimestamp);
-                }
+            // If the snapshot sync apply has completed (determined by the end marker) notify
+            // plugin the completion, so checkpoint/trim process is resumed.
+            if (ack.getMetadata().getMessageMetadataType() == SNAPSHOT_END) {
+                processSnapshotSyncApplied(ack);
             }
             return ack;
         }
     }
 
+    private void processSnapshotSyncApplied(LogReplicationEntry ack) {
```

Comment:
apply has not been done yet, right?  Only transfer has completed?

Reply:
It has been applied already.